### PR TITLE
Automatically rebuild materials from Gradle

### PR DESCRIPTION
**Pull request by romainguy**

### CODE REVIEW

File: android/samples/hello-triangle/app/build.gradle [hunk 1]
```diff
@@ -1,9 +1,69 @@
...
+    @TaskAction
+    void execute(IncrementalTaskInputs inputs) {
+        if (!inputs.incremental) {
+            project.delete(project.fileTree(outputDir).matching { include '*.filamat' })
+        }
+
+        inputs.outOfDate { InputFileDetails outOfDate ->
+            def file = outOfDate.file
+            project.exec {
+                executable "${matcPath}/bin/matc"
```
Comment (bejado) [inline android/samples/hello-triangle/app/build.gradle]: This won't work on Windows, which is fine for now, but we might want to say that somewhere.

File: android/samples/hello-triangle/app/build.gradle [hunk 2]
```diff
@@ -1,9 +1,69 @@
-apply plugin: 'com.android.application'
+// This script accepts the following parameters:
+//
+// filament_tools_dir
+//     Path to the Filament distribution/install directory for desktop
```
Comment (bejado) [inline android/samples/hello-triangle/app/build.gradle]: You could add a similar check to what Tungsten is doing to check for `matc` binary [here](https://github.com/google/filament/blob/master/tools/tungsten/plugin/build.gradle#L38-L41)

File: android/samples/hello-triangle/app/build.gradle (hunk 1 above)
Comment (romainguy) [reply android/samples/hello-triangle/app/build.gradle]: Fixed!

File: android/samples/hello-triangle/app/build.gradle (hunk 2 above)
Comment (romainguy) [reply android/samples/hello-triangle/app/build.gradle]: Done!

**Pull request merged**